Tweaks

DouweM opened the pull request:
I did some refactoring, plus I made the following significant changes:
- Wrap cache read/write calls in notification instrumentation for fragment read/write logging.
- Use correct template when templates are derived using `item.to_partial_path`.

n8 commented:
@DouweM just pulled in some pull requests, and now your work conflicts. Would you mind please updating? Sorry about the late attention to this.

DouweM commented:
Yup, I'll rebase tonight.

DouweM committed:
Clean up Railtie.

DouweM committed:
Check caching being enabled on the view's controller.

DouweM committed:
Wrap cache read/write calls in notification instrumentation for fragment read/write logging and use correct template when templates are derived using item.to_partial_path

@collection_data is an array containing the partial path for every item. Items in @collection need to have a  @collection_data counterpart at the same index.

DouweM commented:
Updated!

n8 commented:
Is the custom key test failing for you?

```
1) MultiFetchFragments works for passing in a custom key
     Failure/Error: view.render(:partial => "views/customer", :collection => [ customer ], :cache => Proc.new{ |item| [item, 'key']}).should == "Hello: david\n"
       expected: "Hello: david\n"
            got: "Hello" (using ==)
       Diff:
       @@ -1,2 +1,2 @@
       -Hello: david
       +Hello
```

DouweM commented:
**Edit**: Disregard this.

It _was_ failing for me before when it read `.should == "Hello"`; it works with the current `.should == "Hello: david\n"`. I'm not sure where `Hello` comes from any way, seeing how [spec/views/_customer.html.erb](https://github.com/n8/multi_fetch_fragments/blob/master/spec/views/_customer.html.erb) reads `Hello: [name]`.

DouweM commented:
Any idea when you can review this?

DouweM committed:
Pass a list instead of an Array to #read_multi.

DouweM commented:
@n8 Ping!

DouweM committed:
Don't call #to_str on a nil result.

DouweM committed:
Don't assume every rendered view has a controller.
